include/drivers/bbram.h

```diff
@@ -0,0 +1,267 @@
+/*
+ * Copyright (c) 2021 Google Inc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_BBRAM_H
+#define ZEPHYR_INCLUDE_DRIVERS_BBRAM_H
+
+#include <device.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * API template to check if the BBRAM is invalid.
+ *
+ * @see z_impl_bbram_get_ibbr
+ */
+typedef int (*bbram_api_get_ibbr)(const struct device *dev);
```

Comment:
What is IBBR? This naming in this API seems too specific to npcx. I think it should use more generic words to describe things.

Reply:
See the referenced function for more details, IBBR stands for invalid battery-backed RAM. From the bit of research I did this should be fairly common to be able to check that the BBR is valid.